libqtile/bar.py

```diff
@@ -273,24 +274,30 @@ def _configure(self, qtile, screen):
             self.window.opacity = self.opacity
             self.window.unhide()
 
-            self.drawer = self.window.create_drawer(width, height)
-            self.drawer.clear(self.background)
-
             self.window.process_window_expose = self.process_window_expose
+            self.window.process_window_expose = self.draw
             self.window.process_button_click = self.process_button_click
             self.window.process_button_release = self.process_button_release
             self.window.process_pointer_enter = self.process_pointer_enter
             self.window.process_pointer_leave = self.process_pointer_leave
             self.window.process_pointer_motion = self.process_pointer_motion
             self.window.process_key_press = self.process_key_press
 
+        # We create a new drawer even if there's already a window to ensure the
+        # drawer is the right size.
+        self.drawer = self.window.create_drawer(width, height)
+        self.drawer.clear(self.background)
+
         self.crashed_widgets = []
         if self._configured:
             for i in self.widgets:
                 self._configure_widget(i)
         else:
             for idx, i in enumerate(self.widgets):
-                if i.configured:
+                # Create a mirror if this widget is being placed on a different bar
+                # We don't do isinstance(i, Mirror) because importing Mirror (at the top)
+                # would give a circular import as libqtile.widget.base imports lbqtile.bar
+                if i.configured and i.bar != self and i.__class__.__name__ != "Mirror":
```

Comment:
Could you use `isinstance(i, widget.Mirror)`? Just seems more canonical

Reply:
Ah, yes. I can, I didn't realise we'd included `Mirror` in `libqtile.widget.__init__.py`. That should hopefully work.

Reply:
Actually, no, this doesn't work as importing `libqtile.widget.__init__.py` imports the `ImportError` widget which is a text box inheriting from `libqtile.widget.base.py` which inherits `bar.py`...

Reply:
Ha, great. Cool, np.